music21/musicxml/partStaffExporter.py

```diff
@@ -438,6 +443,15 @@ def setEarliestAttributesAndClefsPartStaff(self, group: StaffGroup):
         '''
         initialPartStaffRoot: Optional[Element] = None
         mxAttributes: Optional[Element] = None
+        initialM21Key: Optional[KeySignature] = group[0].flat.keySignature
```

Comment:
Check that group is not empty.  A StaffGroup, like any spanner, can have zero items in it.  Especially if someone has been deleting parts.

Reply:
And yes, we should be using `.recurse().getElementsByClass(key.KeySignature)` not because of keySignature, but because `.flat` is really expensive in terms of time and memory.

Btw -- I'm going to add to StreamIterator, a `.first()` method, which returns the first item or None if there are no matching items.  This will help in a lot of places.

Reply:
Yeah, safer to check the group since the function is public (I was lazily depending on empty groups being filtered out by joinableGroups() but better to be safe.)